### PR TITLE
Support weights 0 and 100 in traffic splitting

### PR DESCRIPTION
**Pull request by haywoodsh**
### Proposed changes

Allow split traffic's weight value in virtual server to be 0% or 100%. 

This change is proposed because "0%" is not currently a valid syntax in `split_clients`. When the value is set to 0%, it will not be included in the generated config, but the effect will be the same as if it were. A value of 100% enables full traffic routing to a particular upstream.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]
🚀 
@haywoodsh could you add description to the Proposed Changes section?

File: pkg/apis/configuration/validation/virtualserver_test.go [hunk 1]
```diff
@@ -1583,32 +1583,89 @@ func TestValidatePath(t *testing.T) {
 
 func TestValidateSplits(t *testing.T) {
```
Comment (shaun-nx) [inline pkg/apis/configuration/validation/virtualserver_test.go:1584]: I would add a test to `TestValidateSplitsFails` as well to test that a negative number can't be used.

This might look like this:
```
{
	splits: []v1.Split{
		{
			Weight: 100,
			Action: &v1.Action{
				Pass: "test-1",
			},
		},
		{
			Weight: -2,
			Action: &v1.Action{
				Pass: "test-2",
			},
		},
	},
	upstreamNames: map[string]sets.Empty{
		"test-1": {},
		"test-2": {},
	},
	msg: "negative weight",
},
```

Comment (haywoodsh) [reply pkg/apis/configuration/validation/virtualserver_test.go:1584]: test case added

**Review by shaun-nx — APPROVED** [review]
